src/main/webapp/teacher/controlGameInstance/controlGameInstance.js

```diff
@@ -235,4 +361,87 @@ function initUIControlButtons(gameInstanceId) {
   });
 }
 
+// Probably should do this following the Linked List instead of querySnapshot.forEach
```

Comment:
If you're getting all of the questions, getting the array is probably better : )

Reply:
Gotcha, was thinking about the fact that Firestore stores documents by ordering alphabetically so _maybe_ the array could contain the questions not in the original order? Actually I think this only would happen if we modify the game question's order after we have created it, because at the moment of game creation, Firestore creates documents for each question also in alphabetical order (I think), so maybe that's why we haven't got any issues with the order of question display in the teacher UI...